### PR TITLE
Sds cleanup

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/credentials/kube/secrets.go [hunk 1]
```diff
@@ -100,19 +96,13 @@ func NewCredentialsController(client kube.Client, clusterID cluster.ID) *Credent
...
-
+		secretInformer:     informer,
+		secretLister:       listersv1.NewSecretLister(informer.GetIndexer()),
 		sar:                client.Kube().AuthorizationV1().SubjectAccessReviews(),
-		clusterID:          clusterID,
 		authorizationCache: make(map[authorizationKey]authorizationResponse),
 	}
 }
 
-func toUser(serviceAccount, namespace string) string {
```
Comment (hzxuzhonghu) [inline pilot/pkg/credentials/kube/secrets.go:112]: replace with k8s lib

File: pilot/pkg/credentials/kube/secrets.go [hunk 2]
```diff
@@ -156,19 +146,8 @@ func (s *CredentialsController) insertCache(user string, response error) {
 	}
 }
 
-// DisableAuthorizationForTest makes the authorization check always pass. Should be used only for tests.
-func DisableAuthorizationForTest(fake *fake.Clientset) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/credentials/kube/secrets.go:160]: move to test, which uses it

**Review by ramaraochavali — APPROVED** [review]

File: pilot/pkg/credentials/kube/secrets.go [hunk 3]
```diff
@@ -156,19 +146,8 @@ func (s *CredentialsController) insertCache(user string, response error) {
...
-			Status: authorizationv1.SubjectAccessReviewStatus{
-				Allowed: true,
-			},
-		}, nil
-	})
-}
-
 func (s *CredentialsController) Authorize(serviceAccount, namespace string) error {
-	user := toUser(serviceAccount, namespace)
+	user := sa.MakeUsername(namespace, serviceAccount)
```
Comment (howardjohn) [inline pilot/pkg/credentials/kube/secrets.go:150]: I am not sure we want to bring in a complete new dependency to replace 1 line of code?

Comment (hzxuzhonghu) [reply pilot/pkg/credentials/kube/secrets.go:150]: This is a good question, I am not sure when I am doing this.

But after, the package provides many utilities, istio could use for authz

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/credentials/kube/secrets.go [hunk 4]
```diff
@@ -26,13 +26,11 @@ import (
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
-	"k8s.io/apimachinery/pkg/runtime"
+	sa "k8s.io/apiserver/pkg/authentication/serviceaccount"
```
Comment (costinm) [inline pilot/pkg/credentials/kube/secrets.go:29]: Should we depend on apiserver ?

**Review by costinm — COMMENTED** [review]
We need to be more careful with the dependencies, not sure apiserver package is meant as a library ( i.e. api stability, etc), and we are already having too many deps.

File: pilot/pkg/credentials/kube/secrets.go (hunk 4 above)
Comment (hzxuzhonghu) [reply pilot/pkg/credentials/kube/secrets.go:29]: It is not k/k, this lib is aimed to be used

**Comment by costinm** [comment]
You're right, all good.

BTW - looking at go.mod in the library - it appears they already switched
to opentelemetry for the instrumentation, we should discuss a plan to do
the same, since
the library we use (opencensus) is mostly deprecated.

**Pull request merged**